rs/core/corpora_cli/src/context/mod.rs

```diff
@@ -133,103 +85,82 @@ impl Context {
-        let style = Style::new().bold().green().on_magenta();
-        self.print(message, style);
+        self.print(message, Style::new().bold().green().on_magenta());
     }
 
-    /// Get user input via an editor
-    /// Opens the user's default editor with the provided content
-    /// # Arguments
-    /// * `initial_content` - The initial content to display in the editor.
-    /// # Returns
-    /// The edited content as a `String`.
     pub fn get_user_input_via_editor(&self, initial_content: &str) -> Result<String, String> {
         get_user_input_via_editor(initial_content)
     }
 }
 
 pub fn get_user_input_via_editor(initial_content: &str) -> Result<String, String> {
-    // Create a temporary file
     let temp_file =
         NamedTempFile::new().map_err(|e| format!("Failed to create temp file: {}", e))?;
-
-    // Write the initial content to the temporary file
-    std::fs::write(temp_file.path(), initial_content)
+    fs::write(temp_file.path(), initial_content)
         .map_err(|e| format!("Failed to write to temp file: {}", e))?;
+    let editor = env::var("EDITOR").unwrap_or_else(|_| detect_default_editor());
 
-    // Get the editor from $EDITOR or default to 'vim'
-    let editor = std::env::var("EDITOR").unwrap_or_else(|_| "vim".to_string());
-    let mut command = Command::new(&editor);
+    println!("DEBUG: Using editor: {}", editor);
 
-    // Add flags for non-blocking editors
-    if editor.contains("code") || editor.contains("subl") {
-        command.arg("--wait");
-    } else if editor.contains("gedit") {
-        command.arg("--standalone");
+    let mut command = Command::new(&editor);
+    if editor == "code" || editor == "subl" {
+        command.arg("--wait"); // Append --wait conditionally
     }
-
-    // Open the file in the editor and wait for it to close
     let status = command
-        .arg(temp_file.path())
+        .arg(temp_file.path()) // Always pass the file path separately
         .status()
-        .map_err(|e| format!("Failed to launch editor '{}': {}", editor, e))?;
+        .map_err(|e| {
+            format!(
+                "Failed to launch editor '{} {}': {}",
+                editor,
+                temp_file.path().display(),
+                e
+            )
+        })?;
 
-    // Check if the editor exited successfully
     if !status.success() {
         return Err(format!("Editor '{}' exited with a non-zero status", editor));
     }
+    fs::read_to_string(temp_file.path()).map_err(|e| format!("Failed to read edited file: {}", e))
+}
+
+fn detect_default_editor() -> String {
+    if env::var("VSCODE_PID").is_ok() || env::var("VSCODE_IPC_HOOK_CLI").is_ok() {
+        return "code".to_string(); // No more hardcoded path!
+    }
+    let editors = ["nano", "vim", "vi", "subl", "gedit", "notepad"];
+    for editor in editors {
+        if is_command_available(editor.split_whitespace().next().unwrap()) {
+            return editor.to_string();
+        }
+    }
```

Comment:
No Emacs? 🤯 

There has to be a better way .. but .. interactive mode isn't the future anyways...